lightning-background-processor/src/lib.rs

```diff
@@ -381,7 +380,7 @@ pub async fn process_events_async<
 	OMH: 'static + Deref + Send + Sync,
 	EH: 'static + EventHandler + Send,
 	PS: 'static + Deref + Send,
-	M: 'static + Deref<Target = ChainMonitor<Signer, CF, T, F, L, P>> + Send + Sync,
+	M: 'static + Deref<Target = ChainMonitor<<K::Target as KeysInterface>::Signer, CF, T, F, L, P>> + Send + Sync,
```

Comment:
Should that be part of this PR or in a followup?

Reply:
I actually think there's a big difference between removing Signer from types that were parametrized by both Signer and KeysInterface, and reparametrizing types from Signer to KeysInterface. I'd much prefer to do the latter in a follow-up PR, if possible.

Reply:
If it's ok, I'd rather work a bit more on the Taproot refactors first because I'm starting to be a bit less sure if we wanna convert all the instances from using Signer to using KeysInterface. That's why I'd rather get this out of the way first, but not commit to the rest of the refactor yet.